Building with `make -j$(nproc --ignore 1)` fails non-deterministically

Aha. Here it is:

```
gcc -g -O2  -g -O2 -rdynamic  -o stklos base64.o boolean.o boot.o box.o char.o cond.o cpointer.o dynload.o env.o error.o extend.o ffi.o fixnum.o fport.o gnu-getopt.o hash.o keyword.o lib.o list.o misc.o md5.o number.o object.o parameter.o path.o port.o print.o proc.o process.o promise.o read.o regexp.o signal.o sio.o socket.o sport.o stklos.o str.o struct.o symbol.o syntax.o system.o utf8.o uvector.o vector.o vm.o vport.o thread-common.o thread-pthreads.o mutex-common.o mutex-pthreads.o  -lgmp -lpcre  -lpcreposix -lffi  -lgc -lpthread -ldl  -lm -lpthread 
make[2]: Leaving directory '/home/jeronimo/pkg/scheme/STklos-github/src'
make[1]: Leaving directory '/home/jeronimo/pkg/scheme/STklos-github/src'
Making all in utils
make[1]: Entering directory '/home/jeronimo/pkg/scheme/STklos-github/utils'
./tmpcomp -o stklos-compile stklos-compile.stk
./tmpcomp -o stklos-genlex stklos-genlex.stk
./tmpcomp -o stklos-pp stklos-pp.stk
chmod a+x stklos-script
make[1]: Leaving directory '/home/jeronimo/pkg/scheme/STklos-github/utils'
Making all in lib
make[1]: Entering directory '/home/jeronimo/pkg/scheme/STklos-github/lib'
../src/stklos -q -l srfis.stk \
                 -f ../utils/update-srfi-list.stk \
                     -- --internal > srfis-data.scm
../utils/tmpcomp -o getopt.ostk getopt.stk
WARNING: "getopt" was not provided
**** Error while executing command ("../utils/update-srfi-list.stk" "--internal")

EXIT
make[1]: *** [Makefile:917: srfis-data.scm] Error 70
make[1]: *** Waiting for unfinished jobs....
make[2]: Entering directory '/home/jeronimo/pkg/scheme/STklos-github/src'
make  all-am
make[3]: Entering directory '/home/jeronimo/pkg/scheme/STklos-github/src'
make[3]: Leaving directory '/home/jeronimo/pkg/scheme/STklos-github/src'
make[2]: Leaving directory '/home/jeronimo/pkg/scheme/STklos-github/src'
make[1]: Leaving directory '/home/jeronimo/pkg/scheme/STklos-github/lib'
make: *** [Makefile:429: all-recursive] Error 1
```

There is another one:

```
...
make[1]: Leaving directory '/home/amirouche/tmp/STklos/src'
Making all in utils
make[1]: Entering directory '/home/amirouche/tmp/STklos/utils'
./tmpcomp -o stklos-compile stklos-compile.stk
./tmpcomp -o stklos-genlex stklos-genlex.stk
./tmpcomp -o stklos-pp stklos-pp.stk
chmod a+x stklos-script
make[1]: Leaving directory '/home/amirouche/tmp/STklos/utils'
Making all in lib
make[1]: Entering directory '/home/amirouche/tmp/STklos/lib'
../utils/tmpcomp -o getopt.ostk getopt.stk
make[2]: Entering directory '/home/amirouche/tmp/STklos/src'
../src/stklos -q -l srfis.stk \
                 -f ../utils/update-srfi-list.stk \
                     -- --internal > srfis-data.scm
../utils/tmpcomp -o compfile.ostk compfile.stk
make  all-am
make[3]: Entering directory '/home/amirouche/tmp/STklos/src'
(cd ..; make SUPPORTED-SRFIS)
make[3]: Leaving directory '/home/amirouche/tmp/STklos/src'
make[2]: Leaving directory '/home/amirouche/tmp/STklos/src'
make[2]: Entering directory '/home/amirouche/tmp/STklos'
make[2]: 'SUPPORTED-SRFIS' is up to date.
make[2]: Leaving directory '/home/amirouche/tmp/STklos'
*** Boot 0
WARNING: "compfile" was not provided
**** Error while executing command ("bb.stk" "boot.img0" "instr0")

EXIT
make[1]: *** [Makefile:856: ../src/boot.img] Error 70
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/home/amirouche/tmp/STklos/lib'
make: *** [Makefile:429: all-recursive] Error 1
# 
```

--- a/.github/workflows/ccpp.yml
+++ b/.github/workflows/ccpp.yml
@@ -6,7 +6,7 @@ jobs:
   build:
 
     runs-on: ubuntu-latest
-    
+
     steps:
     - uses: actions/checkout@v1
     - name: timestamp correction
@@ -14,7 +14,7 @@ jobs:
     - name: configure
       run: ./configure
     - name: make
-      run: make
+      run: make -j$(nproc)
     - name: make check
       run: make check
 #    - name: make distcheck

--- a/lib/Makefile.am
+++ b/lib/Makefile.am
@@ -209,7 +209,7 @@ readline-complete.@SH_SUFFIX@: readline-complete.c
 #
 # If ./srfis.stk we must rebuild
 #     - srfis-data.scm (used by SRFI0 and features)
-srfis-data.scm: srfis.stk
+srfis-data.scm: srfis.stk getopt.ostk
 	$(STKLOS_BINARY) -q -l srfis.stk \
 	                 -f ../utils/update-srfi-list.stk \
                      -- --internal > $@

--- a/lib/Makefile.am
+++ b/lib/Makefile.am
@@ -34,6 +34,7 @@ scheme_BOOT = assembler.stk \
           bonus.stk         \
           boot.stk          \
           callcc.stk        \
+          compfile.stk      \
           compiler.stk      \
           computils.stk     \
           date.stk          \

--- a/lib/Makefile.in
+++ b/lib/Makefile.in
@@ -913,7 +913,7 @@ readline-complete.@SH_SUFFIX@: readline-complete.c
 #
 # If ./srfis.stk we must rebuild
 #     - srfis-data.scm (used by SRFI0 and features)
-srfis-data.scm: srfis.stk
+srfis-data.scm: srfis.stk getopt.ostk
 	$(STKLOS_BINARY) -q -l srfis.stk \
 	                 -f ../utils/update-srfi-list.stk \
                      -- --internal > $@

